sonar-db/src/main/java/org/sonar/db/permission/PermissionDao.java

```diff
@@ -143,4 +150,21 @@ public Void apply(@Nonnull List<Long> partitionedComponentIds) {
   private static PermissionMapper mapper(SqlSession session) {
     return session.getMapper(PermissionMapper.class);
   }
+
+  private static class SelectUsersByPermissionAndComponent implements Function<List<Long>, List<UserWithPermissionDto>> {
+    private final PermissionMapper mapper;
+    private final String permission;
+    private final long componentId;
+
+    private SelectUsersByPermissionAndComponent(PermissionMapper mapper, String permission, long componentId) {
+      this.mapper = mapper;
+      this.permission = permission;
+      this.componentId = componentId;
+    }
+
+    @Override
+    public List<UserWithPermissionDto> apply(@Nonnull List<Long> partitionOfIds) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)